src/Eloquent/Collection.php

```diff
@@ -15,10 +15,9 @@ class Collection extends Base
     /**
      * Generate a nested tree.
      *
-     * @param string $childrenRelation
-     * @return static<int, TModel>
+     * @return $this
```

Comment:
This actually returns `static` as opposed to `$this`:

- `self` a new instance of the declaring class
- `static` a new instance of the extending class
- `$this` a subtype of `static`, must return same instance

Reply:
Could you have a look if you can make static work? I cannot